tests/test_code_execution.py

```diff
@@ -36,116 +36,127 @@ def _get_sandbox(sandbox_type):
     return get_sandbox(sandbox_type, host=host)
 
 
-@pytest.mark.parametrize("sandbox_type", ['local', 'piston'])
-def test_multiple_code_blocks(sandbox_type):
-    sandbox = _get_sandbox(sandbox_type)
-
-    code = """
-    a = 1
-    a
-    """
-
-    output, session_id = sandbox.execute_code(code)
-    print(output)
-    assert output == {'process_status': 'completed', 'stderr': '', 'stdout': '1\n'}
-    assert session_id is not None
-
-    code = "a + 5"
-    output, session_id2 = sandbox.execute_code(code, session_id=session_id)
-    assert output == {'process_status': 'completed', 'stderr': '', 'stdout': '6\n'}
-    assert session_id == session_id2
-
-
-@pytest.mark.parametrize("sandbox_type", ['local', 'piston'])
-def test_triple_quotes(sandbox_type):
+@pytest.mark.parametrize(("sandbox_type", "language"), [('local', 'python'), ('local', 'ipython'), ('local', 'pypy3'), ('piston', 'python')])
+def test_triple_quotes(sandbox_type, language):
     sandbox = _get_sandbox(sandbox_type)
     code = '''
-    def my_func():
-        """Test function"""
-        print("asdf")
-    my_func()
+def my_func():
+    """Test function"""
+    print("asdf")
+my_func()
 '''
-    output, session_id = sandbox.execute_code(code)
+    output, _ = sandbox.execute_code(code, language=language)
     assert output == {'process_status': 'completed', 'stderr': '', 'stdout': 'asdf\n'}
-    assert session_id is not None
 
 
-@pytest.mark.parametrize("sandbox_type", ['local', 'piston'])
-def test_multiple_prints(sandbox_type):
+@pytest.mark.parametrize(("sandbox_type", "language"), [('local', 'python'), ('local', 'ipython'), ('local', 'pypy3'), ('piston', 'python')])
+def test_multiple_prints(sandbox_type, language):
     sandbox = _get_sandbox(sandbox_type)
 
     code = """
-    print("1")
-    print("2x3")
+print("1")
+print("2x3")
     """
 
-    output, session_id = sandbox.execute_code(code)
+    output, _ = sandbox.execute_code(code, language=language)
     assert output == {'process_status': 'completed', 'stderr': '', 'stdout': '1\n2x3\n'}
-    assert session_id is not None
 
-    code = "print(2)\n15"
-    output, session_id2 = sandbox.execute_code(code, session_id=session_id)
+    code = "print(2)\nprint(15)"
+    output, _ = sandbox.execute_code(code, language=language)
```

Comment:
We actually want to keep session_id here to check that output of previous cells is not present in the current cell. So I'd recommend creating a separate tests for non-ipython languages